probe-rs/src/architecture/arm/sequences/nxp.rs

```diff
@@ -0,0 +1,291 @@
+use std::{
+    sync::Arc,
+    thread,
+    time::{Duration, Instant},
+};
+
+use crate::{
+    architecture::arm::{
+        ap::{ApAccess, GenericAp, IDR},
+        communication_interface::Initialized,
+        dp::{Abort, Ctrl, DpAccess, Select, DPIDR},
+        ApAddress, ArmCommunicationInterface, DapAccess, DpAddress,
+    },
+    core::CoreRegister,
+    CommunicationInterface, DebugProbeError,
+};
+
+use super::ArmDebugSequence;
+
+pub struct LPC55S69(());
+
+impl LPC55S69 {
+    pub fn new() -> Arc<dyn ArmDebugSequence> {
+        Arc::new(Self(()))
+    }
+}
+
+impl ArmDebugSequence for LPC55S69 {
+    fn debug_port_start(
+        &self,
+        interface: &mut ArmCommunicationInterface<Initialized>,
+        dp: DpAddress,
+    ) -> Result<(), DebugProbeError> {
+        log::info!("debug_port_start");
+
+        interface.write_dp_register(dp, Select(0))?;
```

Comment:
This is mostly copypasted from `DefaultArmSequence`. Does the current design allow calling the impl from `DefaultArmSequence` then do the extra custom stuff? It should, I think this will come up very often.

Reply:
I copied this from the CMSIS-Pack sequence, so it's not really copied from `DefaultArmSequence`, it's more that they are both based on very similar sequences from CMSIS-Packs.

For writing custom sequences, I agree that it would be useful if the default sequence can be extended, instead of having to copy it. I think we could just create a separate function `debug_port_start` with the default implementation, which can then be called by `DefaultArmSequence`, or a custom implementation.

Reply:
Code is identical, with the `enable_debug_mailbox` added. IMO when porting sequences from XML, if we see the vendor has copypasted+extended the default sequence we shouldn't copypaste on our side.